doc/md/guides/orms/typeorm.mdx

```diff
@@ -0,0 +1,428 @@
+    program = [
+        "npx",
+        "@ariga/atlas-provider-typeorm",
+        "load",
+        "--path", "./path/to/entities",
+        "--dialect", "mysql", // mariadb | postgres | sqlite | mssql
+    ]
+}
+
+env "typeorm" {
+    src = data.external_schema.typeorm.url
+    dev = "docker://mysql/8/dev"
+    migration {
+        dir = "file://migrations"
+    }
+    format {
+        migrate {
+            diff = "{{ sql . \"  \" }}"
+        }
+    }
+}
+```
+
+### Script mode
+
+<Tabs>
+<TabItem value="js" label="JavaScript" default>
+
+Create a new file named `load.js` with the following contents:
+
+```js
+#!/usr/bin/env node
+
+const loadEntities = require("@ariga/atlas-provider-typeorm/build/load").loadEntities;
+const EntitySchema = require("typeorm").EntitySchema;
+
+// require typeorm entities you want to load
+const user = new EntitySchema(require("./entities/User"));
+const blog = new EntitySchema(require("./entities/Blog"));
+
+loadEntities("mysql", [user, blog]).then((sql) => {
+  console.log(sql);
+});
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+Create a new file named `load.ts` with the following contents:
+
+```ts
+#!/usr/bin/env ts-node-script
+
+import { loadEntities } from "@ariga/atlas-provider-typeorm/build/load";
+
+// import typeorm entities you want to load
+import { User } from "./entities/User";
+import { Blog } from "./entities/Blog";
+
+loadEntities("mysql", [User, Blog]).then((sql) => {
+  console.log(sql);
+});
+```
+</TabItem>
+</Tabs>
+
+Next, in your project directory, create a new file named `atlas.hcl` with the following contents:
+
+<Tabs>
+<TabItem value="js" label="JavaScript" default>
+
+```hcl
+data "external_schema" "typeorm" {
+    program = [
+        "node",
+        "load.js",
+    ]
+}
+
+env "typeorm" {
+    src = data.external_schema.typeorm.url
+    dev = "docker://mysql/8/dev"
+    migration {
+        dir = "file://migrations"
+    }
+    format {
+        migrate {
+            diff = "{{ sql . \"  \" }}"
+        }
+    }
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```hcl
+data "external_schema" "typeorm" {
+    program = [
+        "npx",
+        "ts-node",
+        "load.ts",
+    ]
+}
+
+env "typeorm" {
+    src = data.external_schema.typeorm.url
+    dev = "docker://mysql/8/dev"
+    migration {
+        dir = "file://migrations"
+    }
+    format {
+        migrate {
+            diff = "{{ sql . \"  \" }}"
+        }
+    }
+}
+```
+</TabItem>
+</Tabs>
+
+## Usage
+
+Atlas supports a [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations)
+workflow, where each change to the database is versioned and recorded in a migration file. You can use the
+`atlas migrate diff` command to automatically generate a migration file that will migrate the database
+from its latest revision to the current TypeORM schema.
+
+Suppose we have the following TypeORM `entities` directory, with two entities `Blog` and `User`:
+
+<Tabs>
```

Comment:
something seems duplicate here?

Reply:
it because Im having tab inside tab, its ok.